src/main/java/dk/cs/aau/huppaal/controllers/HUPPAALController.java

```diff
@@ -855,16 +932,17 @@ private void executeRunConfiguration(RunConfiguration config) {
                 runConfigurationExecuteButtonIcon.setIconLiteral("gmi-stop");
                 runConfigurationExecuteButtonIcon.setIconColor(Color.RED.getColor(Color.Intensity.I300));
                 String s;
-                while((s = stdi.readLine()) != null) {
-                    var finalS = s;
-                    Platform.runLater(() -> CodeAnalysis.addMessage(finalS));
-                }
-                while((s = stde.readLine()) != null) {
-                    var finalS = s;
-                    Platform.runLater(() -> CodeAnalysis.addMessage(new CodeAnalysis.Message(finalS, CodeAnalysis.MessageType.ERROR)));
-                }
-                HUPPAAL.showToast(config.name + " finished("+proc.exitValue()+")");
+                while ((s = stdi.readLine()) != null)
+                    Log.addInfo(config.name, s);
+                while ((s = stde.readLine()) != null)
+                    Log.addError(config.name, s);
+                var exitCode = proc.waitFor();
+                var msg = config.name + " exited with code " + exitCode;
+                Log.addInfo(config.name, msg);
+
+                HUPPAAL.showToast(msg);
             } catch (Exception e) {
+                Log.addError(e.getMessage());
                 HUPPAAL.showToast(e.getMessage());
                 e.printStackTrace();
```

Comment:
I am not sure that we need this anymore, if the error is logged

Reply:
Removed in b14fbe10428bce6c6e20e5d1d6bedb882e943693

We might need some way for developers to detect exceptions though... Maybe a `Log.addException`